port/boards/rv32emu/main.c

```diff
@@ -84,14 +128,18 @@ void *qembd_allocmain(size_t size)
 
 static int poll_event()
 {
-	register int a0 asm("a0") = (uintptr_t) &event;
-	register int a7 asm("a7") = 0xc0de;
-	asm volatile("scall" : "+r"(a0) : "r"(a7));
+    if (event_count <= 0)
+        return 0;
+    event = event_queue.base[event_queue.start++];
+    event_queue.start &= queues_capacity - 1;
+    --event_count;
+
 	if (event.type == MOUSE_MOTION_EVENT) {
 		mouse_movement.x += event.mouse.motion.xrel;
 		mouse_movement.y += event.mouse.motion.yrel;
 	}
-	return a0;
+	
+    return 1;
```

Comment:
Why did you return `1` here?

Reply:
> Why did you return `1` here?

I return an integer here to inform the caller whether or not there are any events in the queue. The return value of `poll_event()` is used in the `qembd_dequeue_key_event()` to determine whether the loop should continue or not.